packages/convenient_test_manager/integration_test/test.dart

```diff
@@ -0,0 +1,131 @@
+
+    if (!isSetup) {
+      await setup(registerVmServiceWrapper: false);
+      getIt.registerSingleton<VmServiceWrapperService>(FakeVmServiceWrapper());
+      isSetup = true;
+      Log.d('myAppGoldenTest', 'setup finished');
+    }
+
+    await tester.pumpWidget(MyApp(theme: theme));
+    await sendTestCalls();
+    await expectLater(find.byType(MyApp),
+        matchesGoldenFile('manager-golden-${theme.name}.png'));
+  });
+}
+
+Future<void> sendTestCalls() async {
+  final mService = getIt.get<ConvenientTestManagerService>();
+
+  final groupID = Counter.next();
+  final testID = Counter.next();
+  const testName = 'Test';
+  ReportItem logEntryItem(List<LogSubEntry> items) => ReportItem(
+      logEntry: LogEntry(
+          testName: testName,
+          id: Counter.next(),
+          subEntries: items.map(
+            (x) => x
+              ..id = Counter.next()
+              ..time = msse64(),
+          )));
+
+  final suite = SuiteInfoProto(groupId: 0.toInt64(), groups: [
+    GroupInfoProto(
+      id: groupID,
+      name: 'Group',
+      parentId: 0.toInt64(),
+      entryIds: [testID],
+    ),
+  ], tests: [
+    TestInfoProto(
+      id: testID,
+      parentId: groupID,
+      name: testName,
+    )
+  ]);
+
+  await mService.reportSingle(ReportItem(setUpAll: SetUpAll()));
+  await mService.reportSingle(ReportItem(suiteInfoProto: suite));
+  await Future<void>.delayed(const Duration(seconds: 5));
+
+  await mService.reportSingle(ReportItem(
+    runnerStateChange: RunnerStateChange(
+      state: TestEntryState(result: 'success', status: 'running'),
+      testName: testName,
+    ),
+  ));
+
+  await mService.reportSingle(logEntryItem(
+      [LogSubEntry(title: 'START', type: LogSubEntryType.TEST_START)]));
+
+  await mService.reportSingle(logEntryItem([
+    LogSubEntry(type: LogSubEntryType.GENERAL_MESSAGE, title: 'START APP')
+  ]));
+  await mService.reportSingle(logEntryItem([
+    LogSubEntry(
+        type: LogSubEntryType.GENERAL_MESSAGE,
+        title: 'TAP',
+        message: 'NonExistentButton')
+  ]));
+  await mService.reportSingle(logEntryItem([
+    LogSubEntry(
+        type: LogSubEntryType.SECTION, title: 'SECTION', message: 'Section')
+  ]));
+  await mService.reportSingle(logEntryItem([
+    LogSubEntry(
+      type: LogSubEntryType.TEST_END,
+      title: 'END',
+    )
+  ]));
+
+  await mService.reportSingle(ReportItem(
+      runnerStateChange: RunnerStateChange(
+          state: TestEntryState(result: 'success', status: 'running'))));
+  await mService.reportSingle(ReportItem(
+      tearDownAll: TearDownAll(
+          resolvedExecutionFilter: ResolvedExecutionFilterProto(
+              allowExecuteTestNames: [testName]))));
+  await mService.reportSingle(ReportItem(
+      runnerStateChange: RunnerStateChange(
+          testName: '(tearDownAll)',
+          state: TestEntryState(status: 'complete', result: 'success'))));
+  Log.d('sendTestCalls', 'waiting for 5 seconds to let the UI build');
```

Comment:
If we are using non-integration tests, then we do not need to wait; instead, we just `tester.pump` and the UI must build